.changeset/seven-geckos-rule.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+fix: fallback and rewrite route matching
```

Comment:
Can you please provide a better/longer description of the fix? Users will read it and won't understand what's been fixed here :)